### PR TITLE
Fix waitForHelixRuns ignoring work item failures

### PR DESCRIPTION
**Pull request by alexperovich**
The waitForHelixRuns function was ignoring work item failures when there are no analyses.

### CODE REVIEW

File: tests/pipeline/pipeline-tests.groovy [hunk 1]
```diff
@@ -151,6 +151,71 @@ stage ('Run Tests') {
                 String url = library(libraryName).jobs.generation.Utilities.calculateGitHubURL('foo/bar')
                 assert url == 'https://github.com/foo/bar' : "Incorrect url for github URL"
             },
+            
+            "vars - waitforHelixRuns - failed work item" : {
+                simpleNode('Windows_NT', 'latest') {
+                    dir 'workItem' {
```
Comment (mmitche) [inline tests/pipeline/pipeline-tests.groovy]: I'm guess here, that you need parens around 'workItem'

**Comment by mmitche** [comment]
@dotnet-bot help (testing)

**Comment by mmitche** [comment]
@dotnet-bot help (testing)

**Comment by alexperovich** [comment]
@mmitche PTAL, I finally got the tests to work.

**Comment by mmitche** [comment]
@alexperovich I think it would also make sense to have a passing test.  In this case, a bad change to waitForHelixRuns would still pass the tests.

**Comment by mmitche** [comment]
@dotnet-bot help (testing)

**Comment by alexperovich** [comment]
@mmitch I added a test for a passing helix run.

**Comment by mmitche** [comment]
Awesome, LGTM

**Pull request merged**